collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderActivity.java

```diff
@@ -93,14 +97,49 @@ protected void onCreate(Bundle savedInstanceState) {
         long[] selectedInstanceIDs = null;
         if (savedInstanceState != null && savedInstanceState.containsKey(TO_SEND)) {
             selectedInstanceIDs = savedInstanceState.getLongArray(TO_SEND);
+
+            // Add optional authentication params
+            if (savedInstanceState.containsKey(ApplicationConstants.BundleKeys.URL)
+                    && savedInstanceState.containsKey(ApplicationConstants.BundleKeys.USERNAME)
+                    && savedInstanceState.containsKey(ApplicationConstants.BundleKeys.PASSWORD)) {
+                url = savedInstanceState.getString(ApplicationConstants.BundleKeys.URL);
+                username = savedInstanceState.getString(ApplicationConstants.BundleKeys.USERNAME);
+                password = savedInstanceState.getString(ApplicationConstants.BundleKeys.PASSWORD);
+            }
         } else {
             // get instances to upload...
             Intent intent = getIntent();
+            Bundle bundle = intent.getExtras();
             selectedInstanceIDs = intent.getLongArrayExtra(FormEntryActivity.KEY_INSTANCES);
+
+            if (bundle != null && bundle.containsKey(ApplicationConstants.BundleKeys.URL)
```

Comment:
When I say "Instead of checking if all 3 are set" in my latest comment, I mean here and when reading from the `savedInstanceState`.